### DISCUSSION TITLE
Support for keep-alive?

### DISCUSSION THREAD

**Discussion by kennerstross** [nuxt/nuxt#17915] [Questions] [answered]
What is the status of (or plan for) supporting keep-alive?

**Comment by alnaum** [comment, marked as answer]
Keep-alive seems to work for pages if you define it in pageMeta:

```
/pages/somePage.vue

<script lang="ts" setup>
definePageMeta({
  keepalive: {},
});
<script>
```

**Reply by kennerstross** [reply]
Thanks for the response, alnaum. I've tried a thousand variations of your suggestion, but nothing works. I wonder if it's because I have ssr=false...

**Reply by alnaum** [reply]
It's strange, maybe a bug. I just checked with ssr true and false set in nuxt.config.ts, and this seems works for me too:
![Screen Shot 2022-02-05 at 11 21 18](https://user-images.githubusercontent.com/46481323/152628872-15539c9d-66fc-4745-ac36-4d1dab8443ac.png)
Page states are preserved.
Test project: https://github.com/alnaum/keep-alive-test

**Reply by kennerstross** [reply]
alnaum - Thank you so much for sharing this code. I really appreciate the help.

Although your code worked, I continued to have problems in my own code, following what seemed to be the same model. Finally, I found the problem: It seems that if your navigation path passes through any components that are NOT keep-alive, then the keep-alive logic breaks. For instance, if page1 and page2 are both keep alive, but the way you navigate between them is through a third component such as MyMenuBar, then MyMenuBar must also be keep-alive. At least that's what appears to be the case for me.

By the way, the image that you shared above... where did that come from?

Thanks,
Kenner

**Reply by alnaum** [reply]
Glad you found this. The screenshot from "Vue.js devtools" extension for Google Chrome: https://chrome.google.com/webstore/detail/vuejs-devtools/ljjemllljcmogpfapbkkighbhhppjdbg

**Reply by Csardelacal** [reply]
Came here to report the same issue, if a page without keep alive is requested the previously visited pages are reloaded when returning.
